language_model/rnn_train.py

```diff
@@ -0,0 +1,104 @@
+import sys
+import sqlite3
+import paddle.v2 as paddle
+import ptb
+import gzip
+
+
+def rnn_net(dict_dim, is_generating=False):
+    # dimension of word vector
+    word_vector_dim = 1024
+    #dimension of hidden unit in GRU Encoder network
+    encoder_size = 1024
+
+    # Encoder
+    src_word_id = paddle.layer.data(
+        name='source_word',
+        type=paddle.data_type.integer_value_sequence(dict_dim))
+    target_word_id = paddle.layer.data(
+        name='target_word',
+        type=paddle.data_type.integer_value_sequence(dict_dim))
+
+    src_embedding = paddle.layer.embedding(
+        input=src_word_id,
+        size=word_vector_dim,
+        param_attr=paddle.attr.ParamAttr(name='_source_word_embedding'))
+
+    src_forward = paddle.networks.simple_gru(
+        input=src_embedding, size=encoder_size)
+    src_backward = paddle.networks.simple_gru(
+        input=src_embedding, size=encoder_size, reverse=True)
```

Comment:
The bidirectional RNN is not necessary and it should be removed. The most serious problem is that the bidirectional RNN cannot be used in sequence generation.